code/game/dna/genes/disabilities.dm

```diff
@@ -115,6 +115,40 @@
 /datum/dna/gene/disability/blindness/New()
 	block=BLINDBLOCK
 
+/datum/dna/gene/disability/colourblindness
+	name="Colourblindness"
+	activation_message="Your perception of colour warps and distorts."
+	deactivation_message ="Your perception of colour returns to normal."
+	instability = -GENE_INSTABILITY_MODERATE
+	disability = COLOURBLIND
+	flags = GENE_EYE_DEPENDENT
+	update = 1
+
+/datum/dna/gene/disability/colourblindness/New()
+	block=COLOURBLINDBLOCK
```

Comment:
In this file, the code above this comment (minus the `flags` and `update`) are all you need for the gene to automatically set/unset the disability. The `activate` and `deactivate` procs below won't be needed, except perhaps for the calls to `update_client_colour`, which is what will eventually check the disability flag. No need to modify eyes at all.